oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthCodeHandler.java

```diff
@@ -384,14 +384,15 @@ private AuthorizationCode generateAuthCode(
         CredentialTrustLevel lowestRequestedCredentialTrustLevel =
                 VectorOfTrust.getLowestCredentialTrustLevel(clientSession.getVtrList());
         var currentCredentialStrength = session.getCurrentCredentialStrength();
-        if (isNull(session.getCurrentCredentialStrength())
-                || lowestRequestedCredentialTrustLevel.compareTo(
-                                session.getCurrentCredentialStrength())
-                        > 0) {
-            session.setCurrentCredentialStrength(lowestRequestedCredentialTrustLevel);
-        }
         if (isNull(currentCredentialStrength)
                 || lowestRequestedCredentialTrustLevel.compareTo(currentCredentialStrength) > 0) {
+            session.setCurrentCredentialStrength(lowestRequestedCredentialTrustLevel);
+        }
+
+        if (configurationService.isCurrentCredentialStrengthInOrchSessionEnabled()
+                && (isNull(currentCredentialStrength)
+                        || lowestRequestedCredentialTrustLevel.compareTo(currentCredentialStrength)
+                                > 0)) {
             orchSessionService.updateSession(
                     orchSession.withCurrentCredentialStrength(lowestRequestedCredentialTrustLevel));
```

Comment:
Open question here about the way we use feature flags: do we need these setters to be feature flagged? I would've thought feature flagging would be more pertinent for getters, or where we're using the getters. Is there some particular testing you want to do at staging with this piece of work?

Reply:
I do agree with that there shouldn't be any change. Not expecting any but being careful and following the pattern as keeping the setters and getters separated so therefore keeping the testing the same. Its more just checking nothing broke as I would have probably paused the pipeline to run an identity journey. Don't think its too much more as it will be used in the getter under the same flag.